src/concurrency/ocf_cache_line_concurrency.c

```diff
@@ -833,9 +833,7 @@ int ocf_req_async_lock_rd(struct ocf_request *req, ocf_req_async_lock_cb cb)
 		req->cache->device->concurrency.cache_line;
 	int lock;
 
-	env_rwsem_down_read(&c->lock);
 	lock = _ocf_req_trylock_rd(req);
```

Comment:
After consulting with Mariusz: looks like there is a potential for deadlock in this change when fastpath wins race with concurrent slowpath and then releases the problematic lock in fastpath rollback without transfering lock ownership to the first waiter.

Reply:
if's fixed now + tests are added to validate